core/src/main/java/org/jdbi/v3/core/statement/Script.java

```diff
@@ -68,21 +68,10 @@ public List<String> getStatements() {
     }
 
     private List<String> splitToStatements(String script) {
-        final List<String> statements = new ArrayList<>();
-        String lastStatement = new SqlScriptParser((t, sb) -> {
-            addStatement(sb.toString(), statements);
-            sb.setLength(0);
-        }).parse(CharStreams.fromString(script));
-        addStatement(lastStatement, statements);
+        ScriptTokenHandler scriptTokenHandler = new ScriptTokenHandler();
+        String lastStatement = new SqlScriptParser(scriptTokenHandler)
+            .parse(CharStreams.fromString(script));
 
-        return statements;
-    }
-
-    private void addStatement(String statement, List<String> statements) {
-        String trimmedStatement = statement.trim();
-        if (trimmedStatement.isEmpty()) {
-            return;
-        }
-        statements.add(trimmedStatement);
+        return scriptTokenHandler.addStatement(lastStatement);
```

Comment:
The semantics here where most statements are handled by the parser, but the last statement is special, is a bit weird.
Should we push the last statement handling into the parse operation?